touchforms/backend/touchcare.py

```diff
@@ -193,7 +195,11 @@ def load_all_objects(self):
                                 criteria=self.additional_filters)
         for c in cases:
             self.put_object(c)
-        self.ids = dict(enumerate(self._objects.keys()))
+        # todo: the sorted() call is a hack to try and preserve order between bootstrapping
+        # this with IDs versus full values. Really we should store a _next_id integer and then
+        # update things as they go into self._objects inside the put_object() function.
+        # http://manage.dimagi.com/default.asp?169413
```

Comment:
didn't bother with the above approach because it seemed more likely to introduce other issues and require more extensive QA. still this is pretty brittle and should get looked at closer in the future.